### PR TITLE
Supports commonly known IDE invisible characters.

### PR DESCRIPTION
**Pull request by HassanTaleb90**
Screenshot 2024-08-11 at 5 31 57 PM

### CODE REVIEW

**Review by HassanTaleb90 — COMMENTED** [review]
Supports commonly known IDE invisible characters.

**Comment by krzyzanowskim** [comment]
That is not the problem introduced by this PR, just remark that drawing invisible characters is kinda broken at the moment, and the performance is subpar.

I'm going to merge these awesome changes, and if you want to think about the outstanding issues, I'm happy to welcome your contribution to make it better.

Screenshot 2024-08-11 at 17 07 14

File: Sources/STTextViewAppKit/STTextLayoutFragment.swift [hunk 1]
```diff
@@ -96,21 +97,45 @@ final class STTextLayoutFragment: NSTextLayoutFragment {
...
+                    let symbolString = String(symbol)
+                    let attributes: [NSAttributedString.Key: Any] = [
+                        .font: font,
+                        .foregroundColor: NSColor.placeholderTextColor
+                    ]
+					
+					let frameRect = CGRect(origin: CGPoint(x: segmentFrame.origin.x - layoutFragmentFrame.origin.x, y: segmentFrame.origin.y - layoutFragmentFrame.origin.y), size: CGSize(width: segmentFrame.size.width, height: segmentFrame.size.height))
+					
+                    let charSize = symbolString.size(withAttributes: attributes)
+					print(frameRect, charSize)
```
Comment (krzyzanowskim) [inline Sources/STTextViewAppKit/STTextLayoutFragment.swift:131]: debug leftover?

**Review by HassanTaleb90 — COMMENTED** [review]
remove print(frameRect, charSize)

File: Sources/STTextViewAppKit/STTextLayoutFragment.swift [hunk 2]
```diff
@@ -96,21 +97,45 @@ final class STTextLayoutFragment: NSTextLayoutFragment {
...
+					default: "\u{00B7}"  // • Default symbol for unspecified whitespace
+					}
+                    
+                    let symbolString = String(symbol)
+                    let attributes: [NSAttributedString.Key: Any] = [
+                        .font: font,
+                        .foregroundColor: NSColor.placeholderTextColor
+                    ]
+					
+					let frameRect = CGRect(origin: CGPoint(x: segmentFrame.origin.x - layoutFragmentFrame.origin.x, y: segmentFrame.origin.y - layoutFragmentFrame.origin.y), size: CGSize(width: segmentFrame.size.width, height: segmentFrame.size.height))
```
Comment (krzyzanowskim) [inline Sources/STTextViewAppKit/STTextLayoutFragment.swift:128]:
```suggestion
					let frameRect = CGRect(origin: CGPoint(x: segmentFrame.origin.x - layoutFragmentFrame.origin.x, y: segmentFrame.origin.y - layoutFragmentFrame.origin.y), size: CGSize(width: segmentFrame.size.width, height: segmentFrame.size.height)).pixelAligned
```

**Review by HassanTaleb90 — COMMENTED** [review]
Verified

**Review by krzyzanowskim — APPROVED** [review]
LGTM

**Pull request merged**